app/src/main/java/com/example/topmovies/MainActivity.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.topmovies
+
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupActionBarWithNavController
+import androidx.navigation.ui.setupWithNavController
+import com.example.topmovies.databinding.ActivityMainBinding
+import com.example.topmovies.fragment.AboutDialogFragment
+import com.example.topmovies.fragment.ToolbarBridge
+
+
+class MainActivity : AppCompatActivity(), ToolbarBridge {
+    private var isLoading = true
+    private lateinit var binding: ActivityMainBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        installSplashScreen().setKeepOnScreenCondition { isLoading }
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setupToolBar()
+        setupNavigationController()
+        isLoading = false
+    }
+
+    override fun showUpButton() {
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun hideUpButton() {
+        supportActionBar?.setDisplayHomeAsUpEnabled(false)
+    }
```

Comment:
Well, I would say that people are used to name it"back" not "up" button.
Not forcing to rename. Just sharing thoughts.